### PR TITLE
guidelines: Added CONTRIBUTIONS.md

### PR DESCRIPTION
**Pull request by dutterbutter**
Added a detail step-by-step process for external developers to begin contributing to our polkadot implementation. It is largely a replication of a Contribution template and other open-source projects.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+**1. Set up Go-pre following the instructions in README.md.**
+
+**2. Fork the Go-pre repo.**
+
+Sign in to your Github account or create a new account if you do not have one already. Then navigate your browser to [https://github.com/ChainSafeSystems/go-pre](https://github.com/ChainSafeSystems/go-pre). In the upper right hand corner of the page, click “fork”. This will create a copy of the Go-pre repo in your account.
+
+**3. Create a local clone of Go-pre.**
+
+```
+$ mkdir -p $GOPATH/src/github.com/ChainSafeSystems
```
Comment (priom) [inline CONTRIBUTING.md]: @dutterbutter I believe with new go 1.11 modules, the project is not dependent on GOPATH anymore.

**Comment by dutterbutter** [comment]
Changes made @noot @ansermino

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+We do not expect all part-time contributors to be experts on all the latest Polkadot documentation, but all contributors should at least be familiarized with the Polkadot Specification fundamentals.
+
+### Core Contributors
+
+Core contributors are remote contractors of ChainSafe Systems and are considered critical team members of our organization. Core devs have all of the responsibilities of part-time contributors plus the majority of the following:
+
+-   Stay up to date on the latest Polkadot research and updates
+-   Monitor github issues and PR’s to make sure owner, labels, descriptions are correct
+-   Formulate independent ideas, suggest new work to do, point out improvements to existing approaches
+-   Participate in code review, ensure code quality is excellent, and have ensure high code coverage
```
[the remark this answers is not in the document]
Comment (dutterbutter) [reply CONTRIBUTING.md]: @ansermino can you elaborate on the admin privileges we do/do not have?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+**3. Create a local clone of Go-pre.**
+
+```
+$ mkdir -p $GOPATH/src/github.com/ChainSafeSystems
+$ cd $GOPATH/src/github.com/ChainSafeSystems
+$ git clone git@github.com:ChainSafeSystems/go-pre.git
+$ cd $GOPATH/src/github.com/ChainSafeSystems/go-pre
+```
+
+**4. Link your local clone to the fork on your Github repo.**
```
[the remark this answers is not in the document]
Comment (dutterbutter) [reply CONTRIBUTING.md]: Other open source projects CONTRIBUTIONS.md or in their WIKI often include not only explaining how to contribute but A Developer's Guide and Code Review Guidelines which get very detailed. See go-ethereum, prymaticlabs/prysm. We can make the actual CONTRIBUTIONS section concise as you describe above (fork, commit, lint, test, open pr with info x,y,z) but it seems the standard practice in all large open source projects to provide this level of detail somewhere in the repo.

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ansermino) [reply CONTRIBUTING.md]: honestly don't recall what I was getting at -- I'm just thinking merging/reviewing PR's, stuff like that

**Pull request merged**